airbyte-integrations/connectors/source-hubspot/source_hubspot/manifest.yaml

```diff
@@ -190,9 +190,80 @@ definitions:
       schema:
         $ref: "#/schemas/email_subscriptions"
 
+  campaigns_stream:
+    $ref: "#/definitions/stream_base"
+    name: campaigns
+    primary_key:
+      - id
+    retriever:
+      $ref: "#/definitions/base_retriever"
+      requester:
+        $ref: "#/definitions/base_requester"
+        path: /email/public/v1/campaigns
+      record_selector:
+        type: RecordSelector
+        extractor:
+          type: DpathExtractor
+          field_path:
+            - campaigns
+      paginator:
+        type: DefaultPaginator
+        page_token_option:
+          type: RequestOption
+          inject_into: request_parameter
+          field_name: offset
+        page_size_option:
+          type: RequestOption
+          inject_into: request_parameter
+          field_name: count
+        pagination_strategy:
+          type: CursorPagination
+          page_size: 500
+          cursor_value: "{{ response.get('offset') }}"
+          stop_condition: "{{ not response.get('hasMore') }}"
+      decoder:
+        type: JsonDecoder
+    incremental_sync:
+      type: DatetimeBasedCursor
+      cursor_field: lastUpdatedTime
+      start_datetime:
+        type: MinMaxDatetime
+        datetime: "{{ format_datetime(config.get('start_date', '2006-06-01T00:00:00Z'), '%ms', '%Y-%m-%dT%H:%M:%SZ') }}"
+        datetime_format: "%ms"
+      datetime_format: "%ms"
+      lookback_window: P{{ config.get('lookback_window', 0) }}D
+      cursor_datetime_formats:
+        - "%ms"
+      is_client_side_incremental: true
+    schema_loader:
+      type: InlineSchemaLoader
+      schema:
+        $ref: "#/schemas/campaigns"
+    transformations:
+      - type: CustomTransformation
+        class_name: source_hubspot.components.AddFieldsFromEndpointTransformation
+        requester:
+          $ref: "#/definitions/base_requester"
+          path: "/email/public/v1/campaigns/{{ stream_slice['parent_id'] }}"
+        record_selector:
+          type: RecordSelector
+          extractor:
+            type: DpathExtractor
+            field_path: []
+        $parameters:
+          # Requester and Selector require stream name as a param.
+          name: campaigns
+      - type: DpathFlattenFields
+        field_path:
+          - counters
+        key_transformation:
+          type: KeyTransformation
+          prefix: counters_
```

Comment:
something interesting i noticed was that for Hubspot's `RecordUnnester`, it looks like in addition to `counters`, we always unnest `properties`.

I could also be interpreting the python code wrong, but it seems like we would need another KeyTransformation for `properties_`. Might be worth double checking what the old code did and if it also flattened `properties` fields

Reply:
yes, RecordUnnester works with properties, but if record doesn't have this field it will be ignored.
campaigns schema doesn't have properties according to the [docs](https://developers.hubspot.com/docs/reference/api/marketing/emails/marketing-emails/v1)
regression [tests](https://github.com/airbytehq/airbyte/actions/runs/14638796358/job/41076060372) also fint (TestDataIntegrity.test_record_schema_match_with_state[passed], TestDataIntegrity.test_all_records_are_the_same_with_state[passed])